WalletWasabi.Fluent/Helpers/StartupHelper.cs

```diff
@@ -62,5 +62,33 @@ private static void StartOnWindowsStartup(bool runOnSystemStartup, string pathTo
 				key.DeleteValue(nameof(WalletWasabi), false);
 			}
 		}
+
+		private static async Task StartOnLinuxStartupAsync(bool runOnSystemStartup)
+		{
+			string pathToDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".config", "autostart");
```

Comment:
I would wrap the body of the method in try-catch and return false on exception and true when it succeeds.

Reply:
We dropped the Try pattern for this feature.
If something goes wrong, exception will be thrown, in `GeneralSettingsViewModel` we catch it, log it, and notify the user that something went wrong.

Reply:
ok

Reply:
I have taken a look at the code in `GeneralSettingsViewModel`. In a case of error, the ToggleButton would still show that the modification was successful because the value of `RunOnSystemStartup` is nowhere set back to the previous state.

```
this.WhenAnyValue(x => x.RunOnSystemStartup)
	.ObserveOn(RxApp.MainThreadScheduler)
	.Skip(1)
	.Subscribe(async runOnStartup =>
	{
		try
		{
			await StartupHelper.ModifyStartupSettingAsync(runOnStartup);
			Services.UiConfig.RunOnSystemStartup = runOnStartup;
		}
		catch (Exception ex)
		{
			Logger.LogError(ex);
			await ShowErrorAsync(Title, "Couldn't save your change, please see the logs for further information.", "Error occured.");
		}
	});
```

Reply:
True,
```
catch (Exception ex)
{
	Logger.LogError(ex);
	RunOnSystemStartup = !runOnStartup;
	await ShowErrorAsync(Title, "Couldn't save your change, please see the logs for further information.", "Error occured.");
}
```
How about this?

Reply:
Setting a value will call again the `await StartupHelper.ModifyStartupSettingAsync(runOnStartup);` If it causes an error again then we are in a forever cycle.

Reply:
That's true.
I did this
```
_runOnSystemStartup = !runOnStartup;
```
and the ToggleButton stays the same in case of error and there is no forever loop.
BUT the second time i toggle the button, it just `switches` without hitting any breakpoint inside the `.Subscribe`.
I don't know what is happening at every second time i click the button.

Reply:
tbh... I can't see a better solution than introducing a new variable... discuss this with @molnard, he usually has some good ideas.

```
this.WhenAnyValue(x => x.RunOnSystemStartup)
	.ObserveOn(RxApp.MainThreadScheduler)
	.Skip(1)
	.Subscribe(async runOnStartup =>
	{
		if (_ignoreNextRunOnSystemStartupChange)
		{
			_ignoreNextRunOnSystemStartupChange = false;
			return;
		}

		try
		{
			await StartupHelper.ModifyStartupSettingAsync(runOnStartup);
			Services.UiConfig.RunOnSystemStartup = runOnStartup;
		}
		catch (Exception ex)
		{
			_ignoreNextRunOnSystemStartupChange = true;
			RunOnSystemStartup = !RunOnSystemStartup;
			Logger.LogError(ex);
			await ShowErrorAsync(Title, "Couldn't save your change, please see the logs for further information.", "Error occured.");
		}
	});
```

Reply:
nit: occured -> occurred

Conceptually, I would expect: "user clicked the button, attempt to do the operation, set RunOnSystemStartup value, now allow Avalonia to redraw". 

Currently, it is more like: "switch the button and if it fails, switch it back" which in theory may lead to a flick effect. On the other hand, exceptions are not expected here I guess.

Reply:
Apply a `ReactiveCommand` on the `ToggleSwitch` and handle the business there.

Reply:
Works like a charm! Thanks for the `ReactiveCommand` suggestion.